### DISCUSSION TITLE
Pathlengths at detector

### DISCUSSION THREAD

**Discussion by Angelos-A** [ankrh/MCmatlab#28] [General]
Hi. I was wondering whether it would be possible to get the pathlengths of the detected photons in each medium in a manner similar to MCX.

**Comment by ankrh** [comment]
Hi Angelos

Although MCmatlab does not directly calculate what you want, you could calculate it yourself if you set a high number of examplePaths to record, use depositionCriteria to only record photons that end up on the detector, and write a small script for parsing through the model.MC.examplePaths array.

model.MC.examplePaths is a 2D array where the first three rows are x, y, z values of the photon packets as they propagate through the voxels of the cuboid and the fourth row is the "weight", the fraction of remaining photons in the packet that have not yet been absorbed. Each column is one recorded position along the trajectory. A position is recorded at each voxel transition and at each scattering event. For each photon packet, you can retrieve its final weight when it hits the detector and calculate the path length that the packet had in each of the layers by adding up path lengths between each pair of points (columns).

Each photon packet begins with two columns of NaN values.

- Anders
